### DISCUSSION TITLE
Question about ESP32 boards and transfer speed.

### DISCUSSION THREAD

**Discussion by n4nier** [awawa-dev/HyperSerialESP32#57] [Q&A] [answered]
This may be a noob'ish question but that is what I am to these esp32 boards...

Before my process of setting everything up (TV LED strips with HyperHDR), I first tested with WLED successfully. After dealing with permissions on my Ubuntu machine running HyperHDR I fell to the following message:
"[LEDDEVICE_ADALIGHT] Could not detect HyperSerialEsp8266/HyperSerialESP32 device"

This led me to believe my board cannot meet the 2MB transfer speed requirement... but I wanted verification on that.

This is the board from aliexpress that I purchased: 
[30Pin ESP32-WROOM-32S 32D ESP 32](https://www.aliexpress.com/item/1005006336964908.html?spm=a2g0o.order_list.order_list_main.29.3d8b1802ZQFDpK)

I noticed this in the specifications but it doesn't mean a ton to me.
"Serial port speed: 115200BPS"

Is this board not sufficient?
If possible can this board work with HyperHDR and WLED (since WLED did work with this board) or is it not able to transfer enough data to sync with what my TV is displaying?

Thanks in advance, I just wanted clarification before I purchased another board, I did attempt to google and research my questions but I feel it did not give me a straight answer.

**Comment by awawa-dev** [comment, marked as answer]
Hi
> "Serial port speed: 115200BPS"

The key is BPS: bit or maybe byte per second.

If it's bit per second actual max speed then it's a joke. Even Arduino board are usually faster. If you are not sure about the product and you dont need wifi just in case (wled could be useful for testing the setup) then choose Pico as generic rp2040 boards are guaranteed to work at ~4Mb (at least I havent met such board that couldn't do that). Also Pico boards are usually well supported on even very strange OS like LG WebOS.

If BPS means byte per second then it's around 921600 baud. Still lower than 2000000 but you can recompile the firmware for such speed (see wiki here).

**Comment by n4nier** [comment]
Thanks a bunch for your response, I appreciate your time and effort you put into this. I've gone ahead and ordered a Pico rp2040 board after reading your HyperSerialPico repo!
